instrumentor/src/index.js

```diff
@@ -57,39 +62,127 @@ const instrumentTouchables = path => {
+  // context to extract the hierarchy, and 'e' so we have context on the event.
+  // 'Heap.autotrackPress(<press type>, this, e)'.
+  const autotrackExpression = t.callExpression(
+    t.memberExpression(t.identifier('Heap'), t.identifier(autotrackMethodName)),
+    [
+      t.stringLiteral(eventType),
+      t.identifier(thisIdentifier),
       t.identifier('e'),
-    ]);
+    ]
+  );
 
-    // Create the expression for calling Heap autotrack. Pass 'this' so we have the component
-    // context to extract the hierarchy, and 'e' so we have context on the event.
-    // 'Heap.autotrackPress(<press type>, this, e)'.
-    const autotrackExpression = t.callExpression(
-      t.memberExpression(t.identifier('Heap'), t.identifier('autotrackPress')),
-      [
-        t.stringLiteral(path.node.key.name),
-        t.identifier('this'),
-        t.identifier('e'),
-      ]
-    );
+  // Function body for tracking the Heap event, then calling the original function.
+  const functionBody = buildFunctionWrapper({
+    AUTOTRACK_EXPRESSION: autotrackExpression,
+    ORIGINAL_FUNCTION_CALL: calledFunction,
+  });
 
-    // Function body for tracking the Heap event, then calling the original function.
-    const functionBody = buildFunctionWrapper({
-      AUTOTRACK_EXPRESSION: autotrackExpression,
-      ORIGINAL_FUNCTION_CALL: calledFunction,
+  // Call the function body with the event parameter 'e'.
+  return t.arrowFunctionExpression([t.identifier('e')], functionBody);
+};
+
+const extendsReactComponent = path => {
+  // By the time we start traversing the AST for Switch.js, the 'Switch' class looks like:
+  //
+  //   var Switch = function (_React$Component) {
+  //     (0, _inherits2.default)(Switch, _React$Component);
+  //
+  //     function Switch() {
+  //       <class body>
+  //     }
+  //
+  //     return Switch;
+  //   }(React.Component);
+  //
+  // so to determine whether this extends 'React.Component', we should check the argument to the
+  // function on the RHS of the variable declarator node for 'React.Component' or 'Component'.
+  // :TODO: (jmtaber129): Consider adding a check for the line that looks like:
+  //   (0, _inherits2.default)(Switch, _React$Component)
+  const reactComponentCandidate =
+    path.node.init && path.node.init.arguments && path.node.init.arguments[0];
+  return (
+    reactComponentCandidate &&
+    ((reactComponentCandidate.object &&
+      reactComponentCandidate.object.name === 'React' &&
+      reactComponentCandidate.property &&
+      reactComponentCandidate.property.name === 'Component') ||
+      reactComponentCandidate.name === 'Component')
+  );
+};
+
+const instrumentSwitch = path => {
+  if (instrumentationMap.has('switch')) {
+    // We already instrumented the switch, so do nothing.
+    return;
+  }
+
+  // The method we want to instrument:
+  // * Is named '_handleChange'
+  // * Has a variable declarator parent named 'Switch'
+  // * Extends 'React.Component'.
+  if (
```

Comment:
What do you think about pulling this (up to about line 167) out into its own function?  Giving it a good name could make it easier to understand what's going on in `instrumentSwitch`

Reply:
Done.  (Also flipped the logic on this condition)